webpack.config.js

```diff
@@ -1,11 +1,22 @@
 const path = require('path');
 
-module.exports = [
-    buildJsFromTo({from: 'plugins/plugin.archive_analytics.js', to: 'BookReader/plugins/plugin.archive_analytics.js'}),
-    buildJsFromTo({from: 'plugins/menu_toggle/plugin.menu_toggle.js', to: 'BookReader/plugins/plugin.menu_toggle.js'}),
-    buildJsFromTo({from: 'plugins/tts/plugin.tts.js', to: 'BookReader/plugins/plugin.tts.js'}),
+const listOfFiles = [
+    'plugins/plugin.archive_analytics.js',
+    'plugins/menu_toggle/plugin.menu_toggle.js',
+    'plugins/tts/plugin.tts.js',
 ];
 
+const compileFiles = async () => {
+    const nestedDirs = new RegExp('/(.*)/');
+    return await listOfFiles.map((filePath) => {
+        const flattenedFilePath = filePath.replace(nestedDirs, '/');
+        return buildJsFromTo({
+            from: filePath,
+            to: `BookReader/${flattenedFilePath}`
+        });
+    });
```

Comment:
- Remove await
- Inline the regex; the variable name self-documents what the regex is.

```suggestion
    return files.map(filePath => {
        const flattenedFilePath = filePath.replace(/\/(.*)\//, '/');
        return buildJsFromTo({from: filePath, to: path.join('BookReader', flattenedFilePath)});
    });
```

Reply:
`the variable name self-documents what the regex is.` - yes it speaks to the output but I think it's important to declare what the regex itself is matching to, even if the regex is simple.

`filePath.replace(nestedDirs, '/');` reads better to me than `filePath.replace(/\/(.*)\//, '/');`

Reply:
:+1: Sounds good to me! Could we change its name to `nestedDirRegex`? Regexes tend to be difficult to name, because they always sound like something else (e.g. is it the name of the directory that's nested? Is it the file's directory?).

Reply:
Yep. I can see that naming complexity. Done.